helmfile/Makefile

```diff
@@ -1,17 +1,25 @@
 .PHONY: build
 build:
-	@for version1 in `cat helmfile-versions | head -n 3`; do \
+	@for version1 in `cat helmfile-versions | sed '1d' | sort -r | rev | uniq -s 2 | rev | head -n 3`; do \
 		helmfile_version=$$(echo $$version1 | tr -d "\n"); \
-		for version2 in `cat helm-versions | head -n 3 `; do \
-			helm_version=`echo "$$version2" | cut -d':' -f1 | tr -d "\n"`; \
-			diff_version=`echo "$$version2" | cut -d':' -f2 | tr -d "\n"`; \
-			kubectl_version=`echo "$$version2" | cut -d':' -f3 | tr -d "\n"`; \
-			version="$$helmfile_version-$$helm_version-$$kubectl_version"; \
+		for version2 in `cat helm-versions | sed '1d' | sort -r | rev | uniq -s 2 | rev | head -n 3`; do \
+			helm_version=`echo "$$version2" | tr -d "\n"`; \
+			for version3 in `cat kubectl-versions | sed '1d' | sort -r | rev | uniq -s 2 | rev | head -n 3`; do \
+				kubectl_version=`echo "$$version3" | tr -d "\n"`; \
+				version="$$helmfile_version-$$helm_version-$$kubectl_version"; \
 				echo "@@@ $$version @@@"; \
-				docker build -t chatwork/`basename $$PWD`:"$$version" --build-arg HELMFILE_VERSION="$$helmfile_version" --build-arg HELM_VERSION="$$helm_version" --build-arg HELM_DIFF_VERSION="$$diff_version" --build-arg KUBECTL_VERSION="$$kubectl_version" .; \
+				if ! curl -sSL https://registry.hub.docker.com/v2/repositories/chatwork/helmfile/tags/| sed -e 's/[][]//g' -e 's/"//g' -e 's/ //g' | tr '},{' '\n' | grep "name:$$version" 2>&1; then \
+					docker build -t chatwork/`basename $$PWD`:"$$version" --build-arg HELMFILE_VERSION="$$helmfile_version" --build-arg HELM_VERSION="$$helm_version" --build-arg KUBECTL_VERSION="$$kubectl_version" .; \
```

Comment:
Get three latest versions for each minor version from the version list.
Combine versions of `kubectl`, `helm`, and `helmfile`, respectively, and build any combinations that do not exist in the registry.